src/lm.jl

```diff
@@ -63,18 +63,18 @@ end
 
 cholfact(x::LinearModel) = cholfact(x.pp)
 
-function fit{LmRespT<:LmResp,LinPredT<:LinPred}(::Type{LinearModel{LmRespT,LinPredT}},
-    X::AbstractMatrix, y::Vector)
+function fit{LmRespT<:LmResp,LinPredT<:LinPred, T<:FP}(::Type{LinearModel{LmRespT,LinPredT}},
+    X::@compat(Union{Matrix{T},SparseMatrixCSC{T}}), y::FPVector)
     rr = LmRespT(y)
-    pp = LinPredT(X)
+    pp = LinPredT(full(X))
     installbeta!(delbeta!(pp, rr.y))
     updatemu!(rr, linpred(pp, 0.0))
     LinearModel(rr, pp)
 end
-function fit(::Type{LinearModel}, X::Matrix, y::Vector)
-    T = promote_type(eltype(X), eltype(y), Float32)
```

Comment:
I think `lm` and `glm` have been diverging slightly for a while. The change I made in that commit made the difference smaller, but I focused on a specific issue with abstract types as fields and not on getting `lm` and `glm` in sync. We should aim for getting them in sync and my guess is that `glm` has the better methods so it's probably fine to delete this method.